### DISCUSSION TITLE
Support parallel execution of tests

### DISCUSSION THREAD

**Discussion by samueloph** [curl/curl#10683] [Ideas]
Considering curl has more than 1500 tests, and that running all of them can take around 30 minutes depending on the builder's specs (that is not an upper limit, it can take even more than that), a lot of time would be saved if the tests could be run in parallel.

**Comment by bagder** [comment]
There are 1583 tests to be exact (counting the http ones as well). Especially when using valgrind they do take a long time.

> a lot of time would be saved if the tests could be run in parallel.

Yes, as has been discussed before and it would be a an awesome improvement. Slightly more work than an afternoon patch though.

**Comment by andy5995** [comment]
Related discussion at https://github.com/curl/curl/discussions/10600#discussioncomment-5107238

**Comment by dfandrich** [comment]
Parallel testing was implemented in #10818. The `-jN` option to runtests.pl is used to enable it, and it will likely be enabled by default at some point.
